### PR TITLE
[BugFix] fix aggregation without rollup rewrite

### LINKED ISSUE #29657: mv with group by rewrite failed
### Steps to reproduce the behavior (Required)
1. CREATE TABLE 

use test_starrocks;
drop  MATERIALIZED VIEW if exists  test_starrocks.test_mv_sr_table_v006_1_unnest_range;
drop table  if exists  test_starrocks.dim_test_sr_table ;
drop table  if exists  test_starrocks.test_sr_table_join ;

create table test_starrocks.dim_test_sr_table (
    fplat_form_itg2 bigint,
    fplat_form_itg2_name string
)DISTRIBUTED BY HASH(fplat_form_itg2)
PROPERTIES (
    "replication_num" = "1"
);

insert into  test_starrocks.dim_test_sr_table(fplat_form_itg2,fplat_form_itg2_name) 
 values 
 (2,'微信'),
 (3,'PC'),
 (5,'QQ'),
 (6,'其他');

create table test_starrocks.dim_test_sr_table_acc_type (
    facct_type bigint,
    facct_type_name string
)DISTRIBUTED BY HASH(facct_type)
PROPERTIES (
    "replication_num" = "1"
);

insert into  test_starrocks.dim_test_sr_table_acc_type(facct_type,facct_type_name) 
 values 
 (1,'A账号类型'),
 (2,'B账号类型');

CREATE TABLE test_starrocks.test_sr_table_join( 
          fdate  int,
          fetl_time BIGINT ,
          facct_type BIGINT ,
          fqqid STRING ,
          fplat_form_itg2 BIGINT  ,
          funit BIGINT  ,
          flcnt BIGINT
 )PARTITION BY range(fdate) (
    PARTITION p1 VALUES [ ("20230702"),("20230703")),
    PARTITION p2 VALUES [ ("20230703"),("20230704")),
    PARTITION p3 VALUES [ ("20230704"),("20230705")),
    PARTITION p4 VALUES [ ("20230705"),("20230706"))
 )
 DISTRIBUTED BY HASH(fqqid)
 PROPERTIES (
    "replication_num" = "1"
);

2. create mv
create MATERIALIZED VIEW test_starrocks.mv_1
DISTRIBUTED BY HASH(fdate,fplat_form_itg2_name)
REFRESH MANUAL
AS 
    select t1.fdate, t2.fplat_form_itg2_name,  count(DISTINCT t1.fqqid) AS index_0_8228, sum(t1.flcnt)as index_xxx
    FROM test_starrocks.test_sr_table_join t1 
    LEFT JOIN test_starrocks.dim_test_sr_table t2
    ON t1.`fplat_form_itg2` = t2.`fplat_form_itg2`
    WHERE t1.`fdate` >= 20230702 and t1.fdate <= 20230705
    GROUP BY  fdate, `fplat_form_itg2_name`;

3. query
explain select   t2.fplat_form_itg2_name,  count(DISTINCT t1.fqqid) AS index_0_8228, sum(t1.flcnt)as index_xxx
    FROM test_starrocks.test_sr_table_join t1 
    LEFT JOIN test_starrocks.dim_test_sr_table t2
    ON t1.`fplat_form_itg2` = t2.`fplat_form_itg2`
    WHERE t1.`fdate` = 20230705
    GROUP BY   `fplat_form_itg2_name`;

this is the case of aggregation without rollup.

### Expected behavior (Required)
rewritten by mv_1
### Real behavior (Required)
not rewritten by mv_1
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by ABingHuang**
Fixes #29657 

1. fix aggregate without rollup rewrite failure because of invalid predicate normalization
2. optimize in predicate rewrite
3. optimize range predicate normalization

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java [hunk 1]
```diff
@@ -499,4 +500,104 @@ public ConstantOperator castTo(Type desc) throws Exception {
...
+                return Optional.of(ConstantOperator.createInt(getInt() + 1));
+            }
+        } else if (type.isBigint()) {
+            if (getBigint() == Long.MAX_VALUE) {
+                return Optional.empty();
+            } else {
+                return Optional.of(ConstantOperator.createBigint(getBigint() + 1));
+            }
+        } else if (type.isLargeint()) {
+            return Optional.of(ConstantOperator.createLargeInt(getLargeInt().add(BigInteger.valueOf(1))));
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java:534]: large int exceeding 2**127-1 overflows.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 20 / 21 (95.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/MvNormalizePredicateRule.java|16|17|94.12%|[64]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java [hunk 2]
```diff
@@ -56,7 +61,44 @@ public ScalarOperator toScalarOperator() {
...
+            if (value.size() > 1) {
+                childPredicates.removeAll(value);
+                // add InPredicateOperator
+                List<ScalarOperator> arguments = Lists.newArrayList();
+                arguments.add(value.get(0).getChild(0));
+                for (ScalarOperator predicate : value) {
+                    if (ScalarOperator.isColumnEqualConstant(predicate)) {
+                        arguments.add(predicate.getChild(1));
+                    } else {
+                        // must be InPredicateOperator
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java:92]: add PreConditions.checkState?

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by satanson — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 83 / 92 (90.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|21|30|70.00%|[513, 518, 519, 522, 537, 539, 545, 553, 560]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ScalarOperator.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/OrRangePredicate.java|31|31|100.00%|[]|

**Review by satanson — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**